src_code/src/controllers/services/Reservation/MakeReservation.js

```diff
@@ -0,0 +1,368 @@
+/** Initialize neccessary modules */
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const helpers = require("../../../utils/interfaces/IHelperFunctions");
+const dbHelpers = require("../../../utils/interfaces/IDBHelperFunctions");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+
+/**
+ * Synchronously processes the request to make a reservation, handling transaction, validation, and insertion.
+ * @param {Object} res - The response object.
+ * @param {Object} req - The request object containing reservation information.
+ * @returns {Object} - Response object indicating the success or failure of the reservation creation.
+ * Request Body: 
+ * {
+ *      "startDate": datetime,
+ *      "endDate": datetime,
+ *      "schoolId": int,
+ *      "reservedEquipments": 
+ *      [
+ *          {
+ *              "modelId": int,
+ *              "typeId": int,
+ *              "quantity": int,
+ *          }
+ *      ]
+ *  }
+ */
+async function MakeReservation(res, req) {
+    /** Start transaction */
+    const trx = await db.transaction();
+
+    try{
+        /** Validate the requested information */
+        const errors = await Promise.resolve(MakeReservationValidation(res, req));
+        if(errors) {
+            return errors;
+        }
+
+        /** Destructure variables from the request body */
+        const { startDate, endDate, schoolId, reservedEquipments } = req;
+        let reservationStatus = "Requested";
+
+        /** Retrieve user information */
+        const user = await dbHelpers.GetUserInfoBySchoolId(db, schoolId);
```

Comment:
sonarlint: redundant `await` on non-Promise object (resolve it?)

Reply:
Thank you, will do this

Reply:
Done!